docs/designs/flatten_odps_key_value_table.md

```diff
@@ -0,0 +1,107 @@
+# Transform the Column with Key-value Pairs String to Wide Table
+This document describes the design to transform the column with key-value pairs string to a wide ODPS table where
+the values of each key is in a each column.
+
+## Motivation
+Generally, the data developers parse logs to features and the number of features may vary. It is convenient to save features using
+key-value pair string as a column in a table. However, it is difficult to analyze features in the table using SQL syntax. So we need to transform
+the column to a wide table. Then we can analyze each features like calculate the mean and variance for the numeric feature. For example, the key-value column is
+
+| features| label|
+|---------|------|
+| age:19,income:1200,education:Master | 1 |
```

Comment:
Is this format used very often in other places as well? I feel like this should usually be ETL's job to clean this up instead of letting ElasticDL to handle it.

Reply:
Yes, It is a preprocess job for analyzing feature statistics before transformations defined in SQLFlow. ElasticDL will not execute it and users should submit the job independently before training. I think we can place modules about preprocess including table transformation and analysis in ElasticDL and I will create a tools folder in ElasticDL to place the codes.

Reply:
Sounds good.